SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h

```diff
@@ -212,6 +222,64 @@ namespace ESpatialOSNetFlow
 	};
 }
 
+UENUM()
+namespace ESpatialOSRuntimeVariant
+{
+	enum Type
+	{
+		Standard,
+		CompatibilityMode
+	};
+}
+
+USTRUCT()
+struct SPATIALGDKEDITOR_API FRuntimeVariantVersion
+{
+	friend class USpatialGDKEditorSettings;
+	friend class FSpatialRuntimeVersionCustomization;
+
+	GENERATED_BODY()
+
+	FRuntimeVariantVersion() = default;
+
+	FRuntimeVariantVersion(const FString& InPinnedVersion) : PinnedVersion(InPinnedVersion)
+	{}
+
+	/** Returns the Runtime version to use for cloud deployments, either the pinned one, or the user-specified one depending on the settings. */
+	const FString& GetVersionForCloud() const;
+
+	/** Returns the Runtime version to use for local deployments, either the pinned one, or the user-specified one depending on the settings. */
+	const FString& GetVersionForLocal() const;
+
+	bool GetUseGDKPinnedRuntimeVersionForLocal() const { return bUseGDKPinnedRuntimeVersionForLocal; }
+
+	bool GetUseGDKPinnedRuntimeVersionForCloud() const { return bUseGDKPinnedRuntimeVersionForCloud; }
+
+	const FString& GetPinnedVersion() const { return PinnedVersion; }
+
+private:
+
+	/** Whether to use the GDK-associated SpatialOS runtime version for local deployments, or to use the one specified in the RuntimeVersion field. */
+	UPROPERTY(EditAnywhere, config, Category = "Runtime", meta = (DisplayName = "Use GDK pinned runtime version for local"))
+	bool bUseGDKPinnedRuntimeVersionForLocal = true;
+
+	/** Whether to use the GDK-associated SpatialOS runtime version for cloud deployments, or to use the one specified in the RuntimeVersion field. */
+	UPROPERTY(EditAnywhere, config, Category = "Runtime", meta = (DisplayName = "Use GDK pinned runtime version for cloud"))
+	bool bUseGDKPinnedRuntimeVersionForCloud = true;
+
+	/** Runtime version to use for local deployments, if not using the GDK pinned version. */
+	UPROPERTY(EditAnywhere, config, Category = "Runtime", meta = (EditCondition = "!bUseGDKPinnedRuntimeVersionForLocal"))
+	FString LocalRuntimeVersion;
+
+	/** Runtime version to use for cloud deployments, if not using the GDK pinned version. */
+	UPROPERTY(EditAnywhere, config, Category = "Runtime", meta = (EditCondition = "!bUseGDKPinnedRuntimeVersionForCloud"))
+	FString CloudRuntimeVersion;
+
+private:
+	/** Pinned version for this variant. */
+	FString PinnedVersion;
```

Comment:
This isn't a UPROPERTY so it can be populated at runtime with whatever the current compiled pinned version in? Are we guaranteed to always set this? That default constructor seems slightly problematic. Can we make it private?

Reply:
Yeah it's not a UPROP so the user cannot change their ini / settings to make the pinned version anything different to the constant we have in source. We always set this in the constructor, you're right about the default constructor then being problematic though. The default constructor must exist and must be public. How about I change the default to always use the `SpatialOSRuntimePinnedStandardVersion` ?

Reply:
I've gone and made the default use `SpatialOSRuntimePinnedStandardVersion`